optimism/src/main.rs

```diff
@@ -135,38 +123,16 @@ pub fn main() -> ExitCode {
 
             // Update the witness with the Keccak step columns before resetting the environment
             // TODO: simplify the contents of the KeccakWitness or create an iterator for it
-            keccak_current_pre_folding_witness
-                .hash_index
-                .push(keccak_env.keccak_witness.hash_index);
-            keccak_current_pre_folding_witness
-                .step_index
-                .push(keccak_env.keccak_witness.step_index);
-            for (env_wit, pre_fold_wit) in keccak_env
-                .keccak_witness
-                .mode_flags
-                .iter()
-                .zip(keccak_current_pre_folding_witness.mode_flags.iter_mut())
-            {
-                pre_fold_wit.push(*env_wit);
-            }
             for (env_wit, pre_fold_wit) in keccak_env
                 .keccak_witness
-                .curr
+                .cols
                 .iter()
-                .zip(keccak_current_pre_folding_witness.curr.iter_mut())
-            {
-                pre_fold_wit.push(*env_wit);
-            }
-            for (env_wit, pre_fold_wit) in keccak_env
-                .keccak_witness
-                .next
-                .iter()
-                .zip(keccak_current_pre_folding_witness.next.iter_mut())
+                .zip(keccak_current_pre_folding_witness.cols.iter_mut())
             {
                 pre_fold_wit.push(*env_wit);
             }
 
-            if keccak_current_pre_folding_witness.step_index.len() == DOMAIN_SIZE {
```

Comment:
Oopsie?

Reply:
GREAT CATCH

Yes, it should be cols[0].len(), will fix that in the next commit